### PR TITLE
Favorites Create Endpoint

### LINKED ISSUE #6: Backend: Favorites Create Request
POST /api/v1/favorites

This end point creates a new favorite for your database. Please note that the rating system should only allow for a number between 1-100.

To create a new favorite, use the following parameters:
```
{
  "favorites": {
    "id": 1,
    "name": "We Will Rock You",
    "artist_name": "Queen"
    "genre": "Rock",
    "rating": 88
  }
}
```
If a favorited song is successfully created, the item will be returned. If the favorite is not successfully created, a 400 status code will be returned. All fields are required.

### PR DESCRIPTION
**Pull request by cebarks**
closes #6 

Features: 
- adds testing for favorites create endpoint
- implements favorites create endpoint

Bug Fix: 
- changes seed files to not specify the ID for row to be inserted, fixes error where knex thinks the id is currently at 1.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,6 +21,41 @@ app.get('/api/v1/favorites', (request, response) => {
     });
 });
 
+app.get('/api/v1/favorites/:id', (request, response) => {
+  database('favorites').where('id', request.params.id).select()
+    .then((favorites) => {
```
Comment (bendelonlee) [inline index.js]: Do you have reasoning for this to be favorites instead of favorite?

File: seeds/playlists.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+exports.seed = function(knex, Promise) {
```
Comment (bendelonlee) [inline seeds/playlists.js:1]: I'm pretty unsure what the keyword Promise is doing in this line

**Review by bendelonlee — CHANGES_REQUESTED** [review]
I have one minor change request. I might let it slide, but it's important to get in practice with this.

File: seeds/playlists.js (hunk 2 above)
Comment (cebarks) [reply seeds/playlists.js:1]: I'm pretty sure it's somehow defining what's expected to be returned?

File: index.js (hunk 1 above)
Comment (cebarks) [reply index.js]: nope not at all, i'll change that real quick.

File: seeds/playlists.js (hunk 2 above)
Comment (bendelonlee) [reply seeds/playlists.js:1]: This article seems to suggest it isn't needed, which I would be inclined to believe:
https://stackoverflow.com/questions/52519636/is-the-promise-argument-passed-into-knex-migrations-needed

**Review by bendelonlee — APPROVED** [review]
Good change

**Pull request merged**